File: Day-15/LongestSubstrinhWithoutRepeatingCharacters.java
import java.util.*;
public class LongestSubstrinhWithoutRepeatingCharacters {
   
    public int lengthOfLongestSubstring(String s) {
        HashMap<Character,Integer> hm=new HashMap<>();
        int max=0;
        int lft=0,rgt=0;
        while(rgt<s.length())
        {
            char c=s.charAt(rgt);
            if(hm.containsKey(c))
            {
                    if(hm.get(c)>=lft)
                    {
                        lft=hm.get(c)+1;
                    }
            }
            max=Math.max(max,rgt-lft+1);
            hm.put(c,rgt);
            rgt++;
        }
        return max;
    
}
       
}

File: Day-15/LongestSubstrinhWithoutRepeatingCharacters.py
class LongestSubstrinhWithoutRepeatingCharacters:
    def lengthOfLongestSubstring(self, s: str) -> int:
        hm={}
        lft=0
        rgt=0
        maxi=0
        while rgt<len(s):
            c=s[rgt]
            if c in hm:
                if hm[c]>=lft:
                    lft=hm[c]+1
            maxi=max(maxi,rgt-lft+1)
            hm[c]=rgt
            rgt+=1
        return maxi
        